assets/app/views/logs/_log_formatted.html

```diff
@@ -0,0 +1,49 @@
+<div>
+  <a
+    name="logTop"
+    id="logTop"></a>
+
+  <div class="log-view">
+    <div
+      ng-show="logList.length > 40"
+      class="log-scroll log-scroll-top"
+      affix
+      offset-top="285"
+      offset-bottom="90">
+      <a ng-click="scrollBottom()">
+        Go to end of log
+      </a>
+    </div>
+    <div layout flex column class="log-view-output">
+      <div class="spinner spinner-inverse" ng-show="!ready"></div>
+      <div
+        layout row
+        class="log-line"
+        ng-repeat="logLine in logList  | filter:logSearch"
+        ng-class-odd="'odd'"
+        ng-class-even="'even'">
+        <a
+          name="line_{{::$index+1}}"
+          id="line_{{::$index+1}}"></a>
+
+        <a class="log-line-number"
+          ng-click="scrollTo('line_'+($index+1), $event)">
+          <span layout flex main-axis="end">
+          {{:: $index+1 }}
+          </span>
+        </a>
+        <span flex class="log-line-text">
+        {{::logLine.text}}
+        </span>
+      </div>
+    </div>
```

Comment:
If we want to do more exploration, I'd lean a towards something (angular native) like [Ionic's Collection Repeater](http://ionicframework.com/docs/api/directive/collectionRepeat/) or [Polymer's iron-list](https://elements.polymer-project.org/elements/iron-list).  There is [ng-infinite-scroll](http://binarymuse.github.io/ngInfiniteScroll/demo_basic.html), but this isn't doing the same kind of view recycling, just adding more and more items at the end as the user scrolls.  [Angular UI scroll](http://angular-ui.github.io/ui-scroll/demo/) still stutters when the list gets long. [Angular UI grid](http://ui-grid.info/) seems overkill.